# Descripción de los datos
# 
Leer los archivos de datos (/datasets/instacart_orders.csv, /datasets/products.csv, /datasets/aisles.csv, /datasets/departments.csv y /datasets/order_products.csv) con pd.read_csv() usando los parámetros adecuados para leer los datos correctamente.

In [5]:
# importar librerías
import pandas as pd
from matplotlib import pyplot as plt

In [6]:
# leer conjuntos de datos en los DataFrames

orders_df = pd.read_csv('../datasets/raw/instacart_orders.csv', sep=';')
products_df = pd.read_csv('../datasets/raw/products.csv', sep=';')
order_product_df= pd.read_csv('../datasets/raw/order_products.csv', sep=';')
aisles_df= pd.read_csv('../datasets/raw/aisles.csv', sep=';')
departaments_df = pd.read_csv('../datasets/raw/departments.csv', sep=';')

### Descripción de los datos

In [9]:
# mostrar información del DataFrame
print('Tabla de Pedidos')
orders_df.info()
print()
print(orders_df.head())
print()
print('Tabla de Productos')
products_df.info()
print()
print(products_df.head())
print()
print('Tabla de Productos por Pedido')
order_product_df.info()
print()
print(order_product_df.head())
print()
print('Tabla de Pasillos')
aisles_df.info()
print()
print(aisles_df.head())
print()
print('Tabla de Departamentos')
departaments_df.info()
print()
print(departaments_df.head())

Tabla de Pedidos
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 478967 entries, 0 to 478966
Data columns (total 6 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   order_id                478967 non-null  int64  
 1   user_id                 478967 non-null  int64  
 2   order_number            478967 non-null  int64  
 3   order_dow               478967 non-null  int64  
 4   order_hour_of_day       478967 non-null  int64  
 5   days_since_prior_order  450148 non-null  float64
dtypes: float64(1), int64(5)
memory usage: 21.9 MB

   order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   1515936   183418            11          6                 13   
1   1690866   163593             5          5                 12   
2   1454967    39980             4          5                 19   
3   1768857    82516            56          0                 20   
4   3007858   196724             2          4    

### Duplicados y valores ausentes

In [ ]:
print('Numero de duplicados:',orders_df.duplicated().sum())
print()
duplicated_orders_mask = orders_df.duplicated()
print(orders_df[duplicated_orders_mask])






Numero de duplicados: 15

        order_id  user_id  order_number  order_dow  order_hour_of_day  \
145574    794638    50898            24          3                  2   
223105   2160484   107525            16          3                  2   
230807   1918001   188546            14          3                  2   
266232   1782114   106752             1          3                  2   
273805   1112182   202304            84          3                  2   
284038   2845099    31189            11          3                  2   
311713   1021560    53767             3          3                  2   
321100    408114    68324             4          3                  2   
323900   1919531   191501            32          3                  2   
345917   2232988    82565             1          3                  2   
371905    391768    57671            19          3                  2   
394347    467134    63189            21          3                  2   
411408   1286742   18322

In [ ]:
print('Numero de duplicados:',products_df.duplicated().sum())
print()
print('Duplicados implicitos:',products_df['product_name'].duplicated().sum())
print()
duplicated_products_mask = products_df['product_name'].duplicated()
print(products_df[duplicated_products_mask])




Numero de duplicados: 0

Duplicados implicitos: 1257

       product_id product_name  aisle_id  department_id
71             72          NaN       100             21
109           110          NaN       100             21
296           297          NaN       100             21
416           417          NaN       100             21
436           437          NaN       100             21
...           ...          ...       ...            ...
49552       49553          NaN       100             21
49574       49575          NaN       100             21
49640       49641          NaN       100             21
49663       49664          NaN       100             21
49668       49669          NaN       100             21

[1257 rows x 4 columns]


In [13]:
print(order_product_df.isna().sum())
print()
print('Numero de duplicados:',order_product_df.duplicated().sum())



order_id               0
product_id             0
add_to_cart_order    836
reordered              0
dtype: int64

Numero de duplicados: 0


In [14]:
print()
print('Numero de duplicados:',aisles_df.duplicated().sum())
print()
print('Duplicados implicitos:',aisles_df['aisle'].duplicated().sum())




Numero de duplicados: 0

Duplicados implicitos: 0


In [15]:
print()
print('Numero de duplicados:',departaments_df.duplicated().sum())
print('Duplicados implicitos:', departaments_df['department'].duplicated().sum())


Numero de duplicados: 0
Duplicados implicitos: 0


1.  orders_df   
En este DataFrame tenemos las siguintes observaciones:

Los nombres de las columnas tienen un buen formato, por lo que no es necesario realizarles algun cambio,
Este DataFrame solamente tiene una columna con datos ausentes por lo que sera necesario revisarlo a detalle más adelante
La columna 'days_since_prior_order' tiene datos de tipo flotante, dado que es un conteo de días, considero que no deberia ser de tipo flotante sino de tipo entero, ya que no hay dias fraccionados.
En el DataFrame existen 15 filas duplicadas, por lo que sugiere algun tipo de error en este día y hora en especifico.

2.  products_df   
En este DataFrame tenemos las siguintes observaciones:

Los nombres de las columnas tienen un buen formato, por lo que no es necesario realizarles algun cambio.
La columna 'product_name' cuenta con datos asusentes o nulos, por lo que sera necesario revisarlo más adelante.
No se encuentran duplicados explicitos, sin embargo al revisar 'product_name' observamos que arrojan duplicados pero por valores ausentes en la misma columna.

3.  order_product_df   
En este DataFrame tenemos las siguintes observaciones:

Los nombres de las columnas tienen un buen formato, por lo que no es necesario realizarles algun cambio.
La columna 'add_to_cart_order' tiene datos de tipo flotante, dado que es un orden de productos, considero que no deberia ser de tipo flotante sino de tipo entero, ya que no hay conteo fraccionado.
el metodo info() no me muestra el conteo de entradas no nulas, sin embargo podemos realizar el conteo con el metodo isna(), ahora ya podemos observar que hay 836 valores nulos en la columna 'add_to_cart_order'

4.  aisles_df   
En este DataFrame tenemos las siguintes observaciones:

Podemos observar que tanto los nombres de las columnas y los tipos de datos son consistentes, y no existen valores nulos en esta DataFrame

5.  departaments_df   
En este DataFrame tenemos las siguintes observaciones:

Podemos observar que tanto los nombres de las columnas y los tipos de datos son consistentes, y no existen valores nulos en esta DataFrame
Como conclusion general, todas los DataFrame creados tienen nombres de columnas consistentes, algunos datos podriamos cambiarlos a otro tipo de dato para que sea congruente con la información, y existen valores nulos que debemos revisar para determinar que realizar con ellos, ademas de que encontramos duplicados explicitos e implicitos que más adelante corregiremos.